### PR TITLE
Update build cache setup to use Gradle Enterprise connector

### PR DESCRIPTION
**Pull request by breskeby**

### CODE REVIEW

File: .ci/init.gradle [hunk 1]
```diff
@@ -50,75 +48,38 @@ final Vault vault = new Vault(
     .engineVersion(1)
     .token(vaultToken)
     .build()
-)
-  .withRetries(5, 1000)
+).withRetries(5, 1000)
 
 
-if (USE_ARTIFACTORY) {
```
Comment (breskeby) [inline .ci/init.gradle:57]: Cleaned that up along the way

File: .ci/init.gradle [hunk 2]
```diff
@@ -50,75 +48,38 @@ final Vault vault = new Vault(
...
   }
 }
 
+void configureGradleEnterprise(def settings) {
+  settings.gradleEnterprise {
+    server = 'https://gradle-enterprise.elastic.co'
+    buildScan.publishAlways()
+  }
 
-final String buildCacheUrl = System.getProperty('org.elasticsearch.build.cache.url')
```
Comment (breskeby) [inline .ci/init.gradle:102]: This explicit configuration is not used anymore using Gradle Enterprise connector. We probably can also remove explicitly passing this in our ci gradle invocations. Or can you see a scenario where having the infra for keeping and reconfiguring makes sense? I vote for just removing the complexity

File: .ci/init.gradle [hunk 3]
```diff
@@ -50,75 +48,38 @@ final Vault vault = new Vault(
...
 gradle.settingsEvaluated { settings ->
   settings.pluginManager.withPlugin("com.gradle.enterprise") {
-    settings.gradleEnterprise {
-      server = 'https://gradle-enterprise.elastic.co'
-    }
+    configureGradleEnterprise(settings)
   }
 }
 
+void configureGradleEnterprise(def settings) {
```
Comment (breskeby) [inline .ci/init.gradle:60]: That configuration block now also works for local remote scripts

File: .ci/init.gradle (hunk 2 above)
Comment (mark-vieira) [reply .ci/init.gradle:102]: I don't think this is required.

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcSetupExtension.java [hunk 4]
```diff
@@ -83,6 +83,11 @@ private TaskProvider<LoggedExec> createRunBwcGradleTask(
                 return getJavaHome(Integer.parseInt(minimumCompilerVersion));
             }));
 
+            // temporally workaround for reworked gradle enterprise setup
+            // removed when PR https://github.com/elastic/elasticsearch/pull/102180 backported
+            // to all BWC branches
+            loggedExec.getEnvironment().put("BUILDKITE", "false");
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcSetupExtension.java:89]: The init script is still properly applying to nested builds though, yes?

**Review by mark-vieira — APPROVED** [review]
LGTM. Do we also need to update [this documentation](https://docs.elastic.dev/elasticsearch-team/onboarding/gradle-setup)?

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcSetupExtension.java (hunk 4 above)
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcSetupExtension.java:89]: I tested the documented init scripts and setup. It should still work with this change

**Pull request merged**